File: Array/beautiful_arrangement_2/python/beautiful_arrangement_2.py
class Solution:
    def constructArray(self, n: int, k: int) -> List[int]:
        if k>=n:
            return []
        temp = [i for i in range(1,n+1)]
        tempSlice = []
        for i in range(k//2+1):
            if temp[i]==temp[k-i]:
                tempSlice.append(temp[i])
            tempSlice.append(temp[i])
            tempSlice.append(temp[k-i])
        for i in range(k+1):
            temp[i]=tempSlice[i]
        return temp

File: Array/beautiful_arrangement_2/c++/beautiful_arrangement_2.cpp
class Solution {
public:
    vector<int> constructArray(int n, int k) {
        if (k>=n){
            return vector<int>{};
        }
        vector<int> temp(n);
        for (int i=0;i<n;i++){
            temp[i]=i+1;
        }
        vector<int> tempSlice;
        for (int i=0 ;i<k/2+1;i++){
            if (temp[i]==temp[k-i]){
                tempSlice.push_back(temp[i]);
            }
            tempSlice.push_back(temp[i]);
            tempSlice.push_back(temp[k-i]);
        }
        for (int i =0;i<k+1;i++){
            temp[i]=tempSlice[i];
        }
        return temp;
    }
};
